octobot_trading/exchanges/exchange_builder.py

```diff
@@ -77,6 +78,11 @@ async def _build_exchange_manager(self):
         # add to global exchanges
         exchanges.Exchanges.instance().add_exchange(self.exchange_manager, self._matrix_id)
 
+    def _init_bot_storage(self):
+        if not storage.RunDatabasesProvider.instance().has_bot_id(self._bot_id):
+            # only one run database per bot id
+            storage.RunDatabasesProvider.instance().add_bot_id(self._bot_id, self.config, self._tentacles_setup_config)
```

Comment:
What is the link with the exchange instance creation?

Reply:
It is only necessary if an exchange with a trading mode is created

Reply:
Could we move this call before calling exchange builder? Is it possible?

Reply:
you mean from OctoBot directly ?

Reply:
We could put it there and in backtesting init process

Reply:
I mean just before calling exchange builder

Reply:
you mean here ? https://github.com/Drakkar-Software/OctoBot-Trading/blob/master/octobot_trading/octobot_channel_consumer.py#L76 I'm not sure I understand what "just before calling exchange builder" means

Reply:
Yes that's where I mean